dashboard/src/components/Config/PkgRepoList/PkgRepoList.tsx

```diff
@@ -99,56 +102,30 @@ function PkgRepoList() {
   }, [allNS, currentNamespace]);
 
   useEffect(() => {
-    const pluginsConfig = {
-      helmGlobalNamespace: helmGlobalNamespace,
-      carvelGlobalNamespace: carvelGlobalNamespace,
-      kubeappsCluster: kubeappsCluster,
-    } as IConfig;
-
-    Promise.all(
-      configuredPlugins
-        .map(plugin =>
-          PackageRepositoriesService.getRepositoriesPermissions(
-            currentNamespace,
-            pluginsConfig,
-            plugin,
-          ),
-        )
-        .filter(i => i !== undefined),
-    ).then(results => {
-      const rbac = new Map<string, IPackageRepositoryPermission>();
-      results.forEach(r => {
-        if (r !== undefined) {
-          rbac.set(JSON.stringify(r.plugin), r);
-        }
-      });
-      setReposRBAC(rbac);
-    });
+    const rbac = new Map<string, PackageRepositoriesPermissions>();
+    reposPermissions.forEach(p => rbac.set(JSON.stringify(p.plugin), p));
+    setReposRBAC(rbac);
 
     // Cluster-wide check
-    Kube.canI(cluster, "kubeapps.com", "apprepositories", "list", "")
-      .then(allowed => setCanSetAllNS(allowed))
-      ?.catch(() => setCanSetAllNS(false));
-  }, [
-    cluster,
-    kubeappsCluster,
-    kubeappsNamespace,
-    helmGlobalNamespace,
-    carvelGlobalNamespace,
-    currentNamespace,
-    configuredPlugins,
-  ]);
+    setCanSetAllNS([...rbac.values()].some(r => r.global["list"]));
+  }, [cluster, reposPermissions]);
 
   const canEditGlobalRepos = (plugin?: Plugin): boolean => {
-    return plugin ? reposRBAC.get(JSON.stringify(plugin))?.global?.update || false : false;
+    if (!plugin) {
+      return false;
+    }
+    return reposRBAC.get(JSON.stringify(plugin))?.global["update"] || false;
   };
 
   const canEditNamespacedRepos = (plugin?: Plugin): boolean => {
-    return plugin ? reposRBAC.get(JSON.stringify(plugin))?.namespaced.update || false : false;
+    if (!plugin) {
+      return false;
+    }
+    return reposRBAC.get(JSON.stringify(plugin))?.namespace["update"] || false;
   };
 
   const canAddRepos = () => {
-    return [...reposRBAC.values()].some(r => r.global?.create || r.namespaced.create);
+    return [...reposRBAC.values()].some(r => r.global["create"] || r.namespace["create"]);
```

Comment:
Perhaps not for this PR, but wondering whether we should move the `"create"`, `"update"`, etc literals to an enum.

Reply:
Agree 🙂

Reply:
Yep, totally agree.